ExampleCode/C++/API/API.cpp

```diff
@@ -626,153 +841,350 @@ bool iSENSE::post_json_email()
-      cout << "object_name.debug()\n";
-      cout << "This will output a ton of data to the console and may help you in\n";
-      cout << "debugging your program.\n";
-
-      curl_easy_cleanup(curl);                // always cleanup
-      curl_global_cleanup();
+  // Check that the project ID is set properly.
+  // When the ID is set, the fields are also pulled down as well.
+  if(project_ID == "empty" || project_ID.empty())
+  {
+    cout << "\nError - please set a project ID!\n";
+    return false;
+  }
 
-      return false;
-    }
+  // Check that a title and contributor key has been set.
+  if(title == "title" || title.empty())
+  {
+    cout << "\nError - please set a project title!\n";
+    return false;
+  }
 
-    cout << "\n\nPOST request failed for some unknown reason.\n";
+  if(email == "email" || email.empty())
+  {
+    cout << "\nErrror - please set an email address!\n";
+    return false;
+  }
 
-    // For cURL return codes, see the following page:
-    // http://curl.haxx.se/libcurl/c/libcurl-errors.html
-    cout << "\ncURL return code was: " << curl_code << endl;
+  if(password == "password" || password.empty())
+  {
+    cout << "\nErrror - please set a password!\n";
+    return false;
+  }
 
-    // Check for errors
-    if(curl_code != CURLE_OK)
-    {
-      fprintf(stderr, "\ncurl_easy_perform() failed: %s\n",
-      curl_easy_strerror(curl_code));
-    }
+  // Make sure the map actually has stuff pushed to it.
+  if(map_data.empty())
+  {
+    cout << "\nMap of keys/data is empty. You should push some data back to this object.\n";
+    return false;
+  }
 
-    curl_easy_cleanup(curl);                // always cleanup
-    curl_global_cleanup();
+  // Set the dataset_ID
+  set_dataset_ID(dataset_ID);
+
+  // Change the upload URL to append rather than create a new dataset.
+  upload_URL = devURL + "/data_sets/append";
+
+  // Call the POST function, give it type 4 since this is append JSON by email & password.
+  int http_code = post_data_function(4);
+
+  /*
+    *  The iSENSE API gives us two response codes to check against:
+    *  Success: 200 OK (iSENSE handled the request fine)
+    *  Failure: 401 Unauthorized (email / password or contributor key was not valid)
+    *  Failure: 422 Unprocessable Entity (email or contributor key was fine,
+    *               but there was an issue with the upload for some reason.)
+    *               the request's formatting. Something in the formatting caused iSENSE to fail.)
+    */
+
+  if(http_code == 200)
+  {
+    cout << "\n\nPOST request successfully sent off to iSENSE!\n";
+    cout << "HTTP Response Code was: " << http_code << endl;
+    cout << "The URL to your project is: " << dev_baseURL << "/projects/" << project_ID << endl;
+    return true;
+  }
+  else if(http_code == 401)
+  {
+    cout << "\nPOST request **failed**\n";
+    cout << "HTTP Response Code was: " << http_code << endl;

```

Comment:
There are a lot of `if (http_code == ###)` in here that seem wrong to be individually checking.  The check for 200 success is fine, but rather than checking 401, 422, or whatever else may be returned, there should be one solid `else` statement that prints the server's error log, and inside here you can do particular checks like (if code == 401, print "make sure contrib key is valid").